src/phpDocumentor/ConfigurationFactory/PhpDocumentor2.php

```diff
@@ -0,0 +1,181 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\ConfigurationFactory;
+
+use phpDocumentor\Dsn;
+
+/**
+ * phpDocumentor2 strategy for converting the configuration xml to an array.
+ */
+final class PhpDocumentor2 implements Strategy
+{
+    /**
+     * Converts the phpDocumentor2 configuration xml to an array.
+     *
+     * @param \SimpleXMLElement $phpDocumentor
+     *
+     * @return array
+     */
+    public function convert(\SimpleXMLElement $phpDocumentor)
+    {
+        $this->validate($phpDocumentor);
+
+        $extensions         = [];
+        $markers            = [];
+        $visibility         = 'public';
+        $defaultPackageName = 'Default';
+        $template           = 'clean';
+        $ignoreHidden       = true;
+        $ignoreSymlinks     = true;
+
+        if (isset($phpDocumentor->parser)) {
+            $extensions = $this->buildExtensionsPart($phpDocumentor->parser);
+            $markers    = $this->buildMarkersPart($phpDocumentor->parser);
+
+            $visibility         = ((string) $phpDocumentor->parser->visibility) ?: $visibility;
+            $defaultPackageName = ((string) $phpDocumentor->parser->{'default-package-name'}) ?: $defaultPackageName;
+            $template           = ((string) $phpDocumentor->transformations->template->attributes()->name) ?: $template;
+
+            if (isset($phpDocumentor->parser->files)) {
+                if (isset($phpDocumentor->parser->files->{'ignore-hidden'})) {
+                    $ignoreHidden = filter_var(
+                        $phpDocumentor->parser->files->{'ignore-hidden'},
+                        FILTER_VALIDATE_BOOLEAN
+                    );
+                }
+
+                if (isset($phpDocumentor->parser->files->{'ignore-symlinks'})) {
+                    $ignoreSymlinks = filter_var(
+                        $phpDocumentor->parser->files->{'ignore-symlinks'},
+                        FILTER_VALIDATE_BOOLEAN
+                    );
+                }
+            }
+        }
+
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $directories     = ((array) $phpDocumentor->parser->files->directory) ?: ['src'];
```

Comment:
The default should not be 'src' but an empty folder (current working directory); this would confuse a use since some projects do not have an src folder